tla/consensus/MCccfraft.tla

```diff
@@ -173,8 +173,8 @@ PostConditions ==
 ----
 \* Refinement
 
-MCRefinementToAbsProp == MappingToAbs(StartTerm..StartTerm + TermCount)!AbsSpec
+MCRefinementToAbsProp == MappingToAbs(StartTerm..StartTerm + TermCount + 1)!AbsSpec
 
-ABSExtend(i) == MappingToAbs(StartTerm..StartTerm + TermCount)!ExtendAxiom(i)
-ABSCopyMaxAndExtend(i) == MappingToAbs(StartTerm..StartTerm + TermCount)!CopyMaxAndExtendAxiom(i)
+ABSExtend(i) == MappingToAbs(StartTerm..StartTerm + TermCount + 1)!ExtendAxiom(i)
+ABSCopyMaxAndExtend(i) == MappingToAbs(StartTerm..StartTerm + TermCount + 1)!CopyMaxAndExtendAxiom(i)
```

Comment:
Adding as a PR comment for now, to not disturb the existing, extremely expensive run. The + 1 is not a general rule, it's really +1 per (primary) retirement.

That's because a retirement, when done by a primary, can lead to a ProposeVoteRequest, which can increment the term by one, _without_ a Timeout (and its gating condition).

There are a few ways this can go:

1. remove the upper limit, like for simulation, if possible, and stop worrying about this
2. improve the gating on terms, by adding an MCRcvProposeVoteRequest with the same restriction as MCTimeout
3. keep the magic +1 and document it

Open to ideas, @lemmy @heidihoward

Reply:
https://github.com/microsoft/CCF/pull/6533

Reply:
#6533 looks good, I think we probably want option 2 as well, the fact that we go past TermCount seems unintentional.

Reply:
> #6533 looks good, I think we probably want option 2 as well, the fact that we go past TermCount seems unintentional.

Also affects https://github.com/microsoft/CCF/pull/6531

Reply:
Option 2 should not, because the runs in #6531 have no reconfiguration, but maybe you mean #6533?